### PR TITLE
Enable avoid_escaping_inner_quotes lint

### LINKED ISSUE #77436: Enable avoid_escaping_inner_quotes lint
1. Comment in this line: https://github.com/flutter/flutter/blob/c9d68ba75f73155ca3882e3a17657022f0ac1a89/analysis_options.yaml#L65
2. Run `flutter analyze --flutter-repo --watch`
3. Fix all warnings printed by previous command
4. Create PR

### PR DESCRIPTION
**Pull request by viralkachhadiya**
Fixes #77436

## Pre-launch Checklist

- [ ] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [ ] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [ ] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [ ] I signed the [CLA].
- [ ] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt.
- [ ] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Comment by Swayam221** [comment]
@viralkachhadiya Hey! looks like you are making the same mistake that I made in my PR #77510 . You've uncommented the line in the analysis_options.yaml alright but haven't fixed the warnings that it produces. Run `flutter analyze` in your local repo and try fixing the info level warnings which say 'avoid escaping inner quotes'.

**Comment by viralkachhadiya** [comment]
@goderbauer

File: dev/benchmarks/complex_layout/ios/Flutter/Debug.xcconfig [hunk 1]
```diff
@@ -1 +1,2 @@
+#include? "Pods/Target Support Files/Pods-Runner/Pods-Runner.debug.xcconfig"
```
Comment (goderbauer) [inline dev/benchmarks/complex_layout/ios/Flutter/Debug.xcconfig:1]: revert this?

**Review by goderbauer — COMMENTED** [review]
Looks liken there are still some failures.

**Comment by goderbauer** [comment]
Looks like some tests are failing. Have you looked at those?

**Comment by viralkachhadiya** [comment]
@jonahwilliams can you help me??

**Comment by jonahwilliams** [comment]
It looks like you've changed some of the test expectations in the tool:

```
05:25 +1664 ~6 -1: test/general.shard/android/deferred_components_gen_snapshot_validator_test.dart: loadingUnitCache validator detects malformed file: not a list [E]                                  
  Expected: <true>
    Actual: <false>
  
  package:test_api                                                                        expect
  test/general.shard/android/deferred_components_gen_snapshot_validator_test.dart 301:5   main.<fn>
  test/general.shard/android/deferred_components_gen_snapshot_validator_test.dart 276:87  main.<fn>
  dart:async                                                                              runZoned
  test/src/common.dart 235:14                                                             testWithoutContext.<fn>
  test/src/common.dart 234:18                                                             testWithoutContext.<fn>
  test/src/common.dart 205:18                                                             test.<fn>
  test/src/common.dart 201:5                                                              test.<fn>
  

05:25 +1664 ~6 -1: test/general.shard/android/deferred_components_gen_snapshot_validator_test.dart: loadingUnitCache validator detects malformed file: missing id                                      
05:25 +1665 ~6 -1: loading test/general.shard/android/gradle_utils_test.dart                                                                                                                           
05:25 +1665 ~6 -1: test/general.shard/android/deferred_components_gen_snapshot_validator_test.dart: loadingUnitCache validator detects malformed file: libraries is list                               
05:25 +1666 ~6 -1: loading test/general.shard/android/gradle_utils_test.dart                                                                                                                           
05:25 +1666 ~6 -1: test/general.shard/android/deferred_components_gen_snapshot_validator_test.dart: loadingUnitCache validator detects malformed file: libraries is list of strings                    
05:25 +1667 ~6 -1: loading test/general.shard/android/gradle_utils_test.dart                                                                                                                           
05:25 +1667 ~6 -1: test/general.shard/android/deferred_components_gen_snapshot_validator_test.dart: loadingUnitCache validator detects malformed file: empty libraries allowed                         
05:25 +1668 ~6 -1: loading test/general.shard/android/gradle_utils_test.dart                                                                                                                           
05:25 +1668 ~6 -1: test/general.shard/android/deferred_components_gen_snapshot_validator_test.dart: androidStringMapping modifies strings file                                                         
05:25 +1669 ~6 -1: loading test/general.shard/android/gradle_utils_test.dart                                                                                                                           
05:25 +1669 ~6 -1: test/general.shard/android/deferred_components_gen_snapshot_validator_test.dart: androidStringMapping adds mapping when no existing mapping                                         
05:25 +1669 ~6 -2: test/general.shard/android/deferred_components_gen_snapshot_validator_test.dart: androidStringMapping adds mapping when no existing mapping [E]                                     
  Expected: <true>
    Actual: <false>
  
  package:test_api                                                                        expect
  test/general.shard/android/deferred_components_gen_snapshot_validator_test.dart 544:5   main.<fn>
  test/general.shard/android/deferred_components_gen_snapshot_validator_test.dart 498:84  main.<fn>
  dart:async                                                                              runZoned
  test/src/common.dart 235:14                                                             testWithoutContext.<fn>
  test/src/common.dart 234:18                                                             testWithoutContext.<fn>
  test/src/common.dart 205:18                                                             test.<fn>
  test/src/common.dart 201:5                                                              test.<fn>
  
  ```

**Comment by viralkachhadiya** [comment]
> It looks like you've changed some of the test expectations in the tool:
> 
> ```
> 05:25 +1664 ~6 -1: test/general.shard/android/deferred_components_gen_snapshot_validator_test.dart: loadingUnitCache validator detects malformed file: not a list [E]                                  
>   Expected: <true>
>     Actual: <false>
>   
>   package:test_api                                                                        expect
>   test/general.shard/android/deferred_components_gen_snapshot_validator_test.dart 301:5   main.<fn>
>   test/general.shard/android/deferred_components_gen_snapshot_validator_test.dart 276:87  main.<fn>
>   dart:async                                                                              runZoned
>   test/src/common.dart 235:14                                                             testWithoutContext.<fn>
>   test/src/common.dart 234:18                                                             testWithoutContext.<fn>
>   test/src/common.dart 205:18                                                             test.<fn>
>   test/src/common.dart 201:5                                                              test.<fn>
>   
> 
> 05:25 +1664 ~6 -1: test/general.shard/android/deferred_components_gen_snapshot_validator_test.dart: loadingUnitCache validator detects malformed file: missing id                                      
> 05:25 +1665 ~6 -1: loading test/general.shard/android/gradle_utils_test.dart                                                                                                                           
> 05:25 +1665 ~6 -1: test/general.shard/android/deferred_components_gen_snapshot_validator_test.dart: loadingUnitCache validator detects malformed file: libraries is list                               
> 05:25 +1666 ~6 -1: loading test/general.shard/android/gradle_utils_test.dart                                                                                                                           
> 05:25 +1666 ~6 -1: test/general.shard/android/deferred_components_gen_snapshot_validator_test.dart: loadingUnitCache validator detects malformed file: libraries is list of strings                    
> 05:25 +1667 ~6 -1: loading test/general.shard/android/gradle_utils_test.dart                                                                                                                           
> 05:25 +1667 ~6 -1: test/general.shard/android/deferred_components_gen_snapshot_validator_test.dart: loadingUnitCache validator detects malformed file: empty libraries allowed                         
> 05:25 +1668 ~6 -1: loading test/general.shard/android/gradle_utils_test.dart                                                                                                                           
> 05:25 +1668 ~6 -1: test/general.shard/android/deferred_components_gen_snapshot_validator_test.dart: androidStringMapping modifies strings file                                                         
> 05:25 +1669 ~6 -1: loading test/general.shard/android/gradle_utils_test.dart                                                                                                                           
> 05:25 +1669 ~6 -1: test/general.shard/android/deferred_components_gen_snapshot_validator_test.dart: androidStringMapping adds mapping when no existing mapping                                         
> 05:25 +1669 ~6 -2: test/general.shard/android/deferred_components_gen_snapshot_validator_test.dart: androidStringMapping adds mapping when no existing mapping [E]                                     
>   Expected: <true>
>     Actual: <false>
>   
>   package:test_api                                                                        expect
>   test/general.shard/android/deferred_components_gen_snapshot_validator_test.dart 544:5   main.<fn>
>   test/general.shard/android/deferred_components_gen_snapshot_validator_test.dart 498:84  main.<fn>
>   dart:async                                                                              runZoned
>   test/src/common.dart 235:14                                                             testWithoutContext.<fn>
>   test/src/common.dart 234:18                                                             testWithoutContext.<fn>
>   test/src/common.dart 205:18                                                             test.<fn>
>   test/src/common.dart 201:5                                                              test.<fn>
>   
> ```

How ??

**Comment by jonahwilliams** [comment]
That is something you'll have to figure out if you want to land this change

**Pull request closed without merging**